woocommerce/woocommerce-gutenberg-products-block
# Change some snake cased properties to camelCase - mostly related to Shipping and Totals

This PR changes the casing of properties in the shipping rates section of the cart from `snake_case` to `camelCase`. The cart_totals property was also changed to be camel cased.

- Convert `has-shipping-rate` to TypeScript
- Convert `derive-selected-shipping-rates` to TypeScript
- Create `mapKeysToCamelCase` function and unit tests for it. This function is designed to change any object's keys to camel case, even nested objects within it. It leaves non-plain objects alone, but does check items in an array, and converts them too if any members are a plain object.
- Add `isPlainObject` type guard. This differentiates between things like `Map`, `Array`, etc.
- Add tests for it.
- Update JS unit tests to account for changes in casing in the dummy data
- Add a new `PreviewCart` type - this is a hybrid of the `Cart` and `PreviewCart` types, when we finish converting all of the properties to camel case this can be removed in favour of simply using `Cart`.
- Add CSS fix for image sizes in the checkout editor.


Fixes #4150 and fixes #4250




### How to test the changes in this Pull Request:

1. Please extensively smoke test the cart and checkout front-end _and_ editor screen.
2. Ensure you test lots of different combinations of tax/shipping/payment settings.
3. Ensure the correct currency is displayed when in the cart and checkout.
3. Ensure you are billed the correct amount when paying.
